### PR TITLE
#166187853 Implement descriptive validation erros messages

### PR DESCRIPTION
**Pull request by jennizalwango**
#### What does this PR do?
- Users should receive a descriptive error message when they enter invalid user parameters in an attempt to signup.

#### Description of Task to be completed?
- User should get descriptive validation errors messages on registration when they enter invalid user parameters when signing up.

#### How should this be manually tested?
Run git fetch
Then check out `ft-signup-validations-166187853`
Then using postman run this `http://127.0.0.1:8000/api/users/ `providing wrong data

#### What are the relevant pivotal tracker stories?
[Finishes #166187853]

#### Screenshots (if appropriate)?
Screenshot 2019-06-10 at 1 14 08 PM

### CODE REVIEW

**Review by Emmanuelaaron — CHANGES_REQUESTED** [review]
Thanks @jennizalwango , though i'm getting unexpected outputs by django not the expected error messages you've wriiten in your code. Can you please look into this and verify it? Thanks

**Comment by alexxsanya** [comment]
> Thanks @jennizalwango , though i'm getting unexpected outputs by django not the expected error messages you've wriiten in your code. Can you please look into this and verify it? Thanks

@jennizalwango awaiting on this to proceed with reviewing this PR.

**Comment by jennizalwango** [comment]
> Thanks @jennizalwango , though i'm getting unexpected outputs by Django not the expected error messages you've wriiten in your code. Can you please look into this and verify it? Thanks
l have implemented this @Emmanuelaaron

**Comment by jennizalwango** [comment]
> > Thanks @jennizalwango , though i'm getting unexpected outputs by django not the expected error messages you've wriiten in your code. Can you please look into this and verify it? Thanks
> 
> @jennizalwango awaiting on this to proceed with reviewing this PR.

l have worked on this @alexxsanya

**Review by HabibSentongo — APPROVED** [review]
I have tested both on localhost and on heroku review app and I'm convinced everything works well now even considering the feedback the team had given. 
You have a great coverage, good job. 👍

File: authors/settings.py [hunk 1]
```diff
@@ -189,4 +189,4 @@
 EMAIL_USE_TLS = os.getenv('EMAIL_USE_TLS')
 EMAIL_PORT = os.getenv('EMAIL_PORT')
 EMAIL_HOST_USER = os.getenv('EMAIL_HOST_USER')
-EMAIL_HOST_PASSWORD = os.getenv('HOST_PASSWORD')
\ No newline at end of file
+EMAIL_HOST_PASSWORD = os.getenv('EMAIL_HOST_PASSWORD')
```
Comment (Emmanuelaaron) [inline authors/settings.py:192]: Great Job @jennizalwango however, why did you have to change `EMAIL_HOST_PASSWORD` environment variable?

It's throwing errors 
Screenshot 2019-06-11 at 08 46 20

**Review by Emmanuelaaron — CHANGES_REQUESTED** [review]

**Review by neelxie — APPROVED** [review]
Good stuff @jennizalwango the feature works well on my end.

File: authors/settings.py (hunk 1 above)
Comment (jennizalwango) [reply authors/settings.py:192]: this has been cleared, it   was about the Heroku environment variable on my review app but all is fine now

**Review by Emmanuelaaron — APPROVED** [review]
Great job @jennizalwango .

**Review by alexxsanya — APPROVED** [review]
Well done @jennizalwango

**Pull request merged**